v3/lints/rfc/lint_ext_cert_policy_disallowed_any_policy_qualifier.go

```diff
@@ -49,16 +57,113 @@ func NewUnrecommendedQualifier() lint.LintInterface {
 }
 
 func (l *unrecommendedQualifier) CheckApplies(c *x509.Certificate) bool {
-	return util.IsExtInCert(c, util.CertPolicyOID)
+
+	// TODO? extract to util method: HasAnyPolicyOID(c)
+	if !util.IsExtInCert(c, util.CertPolicyOID) {
+		return false
+	}
+
+	for _, policyIds := range c.PolicyIdentifiers {
+		if policyIds.Equal(util.AnyPolicyOID) {
+			return true
+		}
+	}
+	return false
 }
 
 func (l *unrecommendedQualifier) Execute(c *x509.Certificate) *lint.LintResult {
-	for _, firstLvl := range c.QualifierId {
-		for _, qualifierId := range firstLvl {
-			if !qualifierId.Equal(util.CpsOID) && !qualifierId.Equal(util.UserNoticeOID) {
+
+	var err, certificatePolicies = getCertificatePolicies(c)
+
+	if err != nil {
+		return &lint.LintResult{Status: lint.Fatal, Details: err.Error()}
+	}
+
+	for _, policyInformation := range certificatePolicies {
+
+		if !policyInformation.policyIdentifier.Equal(util.AnyPolicyOID) { // if the policyIdentifier is not anyPolicy do not examine further
+			continue
+		}
+
+		if len(policyInformation.policyQualifiersBytes.Bytes) == 0 { // this policy information does not have any policyQualifiers
+			continue
+		}
+
+		var policyQualifiersSeq, policyQualifierInfoSeq asn1.RawValue
+
+		empty, err := asn1.Unmarshal(policyInformation.policyQualifiersBytes.Bytes, &policyQualifiersSeq)
+
+		if err != nil || len(empty) != 0 || policyQualifiersSeq.Class != 0 || policyQualifiersSeq.Tag != 16 || !policyQualifiersSeq.IsCompound {
+			return &lint.LintResult{Status: lint.Fatal, Details: "policyExtensions: Could not unmarshal policyQualifiers sequence."}
+		}
+
+		//iterate over policyQualifiers ... SEQUENCE SIZE (1..MAX) OF PolicyQualifierInfo OPTIONAL
+		for policyQualifierInfoSeqProcessed := false; !policyQualifierInfoSeqProcessed; {
+			// these bytes belong to the next PolicyQualifierInfo
+			policyQualifiersSeq.Bytes, err = asn1.Unmarshal(policyQualifiersSeq.Bytes, &policyQualifierInfoSeq)
+			if err != nil || policyQualifierInfoSeq.Class != 0 || policyQualifierInfoSeq.Tag != 16 || !policyQualifierInfoSeq.IsCompound {
+				return &lint.LintResult{Status: lint.Fatal, Details: "policyExtensions: Could not unmarshal policy qualifiers"}
+			}
+			if len(policyQualifiersSeq.Bytes) == 0 { // no further PolicyQualifierInfo exists
+				policyQualifierInfoSeqProcessed = true
+			}
+
+			var policyQualifierId asn1.ObjectIdentifier
+			_, err = asn1.Unmarshal(policyQualifierInfoSeq.Bytes, &policyQualifierId)
+			if err != nil {
+				return &lint.LintResult{Status: lint.Fatal, Details: "policyExtensions: Could not unmarshal policyQualifierId."}
+			}
+
+			if !policyQualifierId.Equal(util.CpsOID) && !policyQualifierId.Equal(util.UserNoticeOID) {
 				return &lint.LintResult{Status: lint.Error}
 			}
 		}
 	}
+
 	return &lint.LintResult{Status: lint.Pass}
 }
+
+func getCertificatePolicies(c *x509.Certificate) (error, []policyInformation) {
```

Comment:
Errors typically come second in the return list.

```suggestion
func getCertificatePolicies(c *x509.Certificate) ( []policyInformation, error) {
```